### PR TITLE
Make container events belong to their container groups

### PR DESCRIPTION
**Pull request by zeari**
BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1496179

part of https://github.com/ManageIQ/manageiq/pull/16583

@cben @moolitayer @enoodle Please review

### CODE REVIEW

**Comment by gmcculloug** [comment]
@lfu Please review

**Comment by lfu** [comment]
LGTM 👍

File: content/automate/ManageIQ/System/Event/EmsEvent/Kubernetes.class/pod_containercreated.yaml [hunk 1]
```diff
@@ -4,9 +4,11 @@ version: 1.0
 object:
   attributes:
     display_name: 
-    name: CONTAINER_FAILED
+    name: POD_CONTAINERCREATED
```
Comment (moolitayer) [inline content/automate/ManageIQ/System/Event/EmsEvent/Kubernetes.class/pod_containercreated.yaml]: I can't find where this event is created?

Comment (zeari) [reply content/automate/ManageIQ/System/Event/EmsEvent/Kubernetes.class/pod_containercreated.yaml]: Looks like a weird git error. 
container_created and container_failed got criss-crossed in file change rename

`Kubernetes.class/container_created.yaml → ...Kubernetes.class/pod_containerfailed.yaml`
`Kubernetes.class/container_failed.yaml     → ...Kubernetes.class/pod_containercreated.yaml`

Comment (moolitayer) [reply content/automate/ManageIQ/System/Event/EmsEvent/Kubernetes.class/pod_containercreated.yaml]: OK didn't found where the POD prefix is being added. @zeari [located it]( https://github.com/ManageIQ/manageiq-providers-kubernetes/blob/3345798e7b1ac541e200beac99e546ffafc606ce/app/models/manageiq/providers/kubernetes/container_manager/event_catcher_mixin.rb#L107)

**Comment by zeari** [comment]
@miq-bot add_label gaprindashvili/yes

**Comment by zeari** [comment]
@miq-bot add_label fine/yes

**Comment by cben** [comment]
>
> @cben <https://github.com/cben> so adding the policies, we will likely
> see a lot of failures with target not found in log.
>
Yes but same is true of existing Pod & Replicator policies support :-(
They are nearly unusable.
We need to add these events, and independently we need to solve
event-before-refresh.

**Comment by zeari** [comment]
@lfu @gmcculloug Please merge.

**Comment by simaishi** [comment]
Gaprindashvili backport details:

```
$ git log -1
commit 36ef4dc36493d592d8d60256ab42498d72808623
Author: Greg McCullough <gmccullo@redhat.com>
Date:   Thu Dec 21 11:37:02 2017 -0500

    Merge pull request #225 from zeari/container_to_pod
    
    Make container events belong to their container groups
    (cherry picked from commit 1696ac15e4d25e94015819b11d36c3c490226531)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1530651
```

**Comment by zeari** [comment]
@simaishi how do `fine` backports work in relation to this change? Do I need to create a PR?

**Comment by simaishi** [comment]
@zeari  If there is a conflict with cherry-pick, I will request a separate PR. Until then, nothing is needed.

**Comment by simaishi** [comment]
Fine backport details:

```
$ git log -1
commit c57a0175b40bbac87a51dde9a9b019fb663da9f9
Author: Greg McCullough <gmccullo@redhat.com>
Date:   Thu Dec 21 11:37:02 2017 -0500

    Merge pull request #225 from zeari/container_to_pod
    
    Make container events belong to their container groups
    (cherry picked from commit 1696ac15e4d25e94015819b11d36c3c490226531)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1530653
```

**Pull request merged**